### ISSUE TITLE
Stamina System + Fire + Space Wind + Getting up = Permastunned for 4 full minutes.

### ISSUE THREAD

**Issue by BurgerLUA** [citadel-station/citadel-main#9313]
## Replication
1. Be stuck in a situation where spacewind forces you against a wall or a grate.
2. Have low stamina.
3. Have another corpse, on fire, be pushed and forced against you.
4. Be stuck on fire for a near-unlimited time because the following things are happening:
- The corpse is constantly pushing against you, transfering fire in a way that violates the laws of thermodynamics.
- You can't force yourself up because of low stamina, and you can't resist get up because "Resist" also means stop drop and roll.
- Space wind is pushing you against the wall.

I mean it's a hilarious situation and quite specific but it exposes 3 problems:
- How fire spreads to people.
- How stamina affects players.
- How spacewind affects players.

Fire spreading is very powerful compared to most servers. It's not meant to be realistic, but meant to make lighting people on fire more viable. When combined with someone else and spacewind, it is actually possible to have a REALLY long fire be transferred constantly between two players because space wind forces a move call every time, which means that even if you're next to someone, it will transfer because the game thinks that the someone bumped into you; even lying down.

tl;dr three bugs present:
- Getting up while constantly being set on fire without having auto-getup on is hard.
- Spacewind calls move procs even though it can't move. Given the frequency of spacewind, it seems a little excessive.
- It is possible to have a near unlimited fire because of how fire is spread between two people.

**Comment by Ghommie** [comment]
weird, do_after(), do_mob() et similar shouldn't fail when attempting to move (and successively bump) onto an inaccessible tile rather than the user's or target's location being effectively changed, along the other factors, iirc. 

Overall, while the nature of the three "bugs" is more of punishing game design than effective bugs, 1 and 3 are legit. 

About issue number 2: `Move()` proc is called when _attempting_ to move a movable atom to a given location, it's an implemented byond proc which also also calls other procs such as `Crossed()` and `Enter()` amongst others which can alter its outcome. It doesn't have to be necessarily used only in cases where it would return TRUE. Most importantly, precautionary `CanMove()` checks will alter how spacewind works (infamous example being electrified grilles not shocking wannabe perma escapists anymore), and add further performance drag to atmos.

**Comment by BurgerLUA** [comment]
I wouldn't say 1 and 3 are legit. 1 isn't legit because getting up and stop-drop-and-roll share the same button, which isn't good game design. 3 isn't legit because fire should not spread like this unless you have fuel or something on you.

**Comment by Ghommie** [comment]
Apologises, by legit i meant a legit issue (that should be resolved).
EDIT: also, by `CanMove()` I actually meant `CanPass()`, but incorrectly called it as such.

**Comment by BurgerLUA** [comment]
OH yeah sorry I completely misread what you said and had a brainfart. Disregard everything I said in my comment, what you said is agreeable.

**Comment by Ghommie** [comment]
Slapping a "Needs Reproducing" Tag here because I don't recall such an unfortunate combination of events permastunning somone rather than effectively killing them on my watch.
